invertedai/large/initialize.py

```diff
@@ -309,45 +309,76 @@ def _insert_agents_into_nearest_regions(
+            recurrent_states = recurrent_states + [recurr for i, recurr in enumerate(response.recurrent_states) if region_agent_keep_map[ind][i]]
+            if get_infractions:
+                infractions = infractions + [infr for i, infr in enumerate(response.infractions) if region_agent_keep_map[ind][i]]
+        
+        response.infractions = infractions
+        response.agent_states = agent_states
+        response.agent_properties = agent_properties
+        response.recurrent_states = recurrent_states 
+    else:
+        raise InvertedAIError(message=f"Unable to initialize any given region. Please check the input parameters.")
+
+    return response
+
 def _get_all_existing_agents_from_regions(
     regions: List[Region],
     exclude_index: Optional[int] = None,
     nearby_region: Optional[Region] = None,
-    region_map: Optional[List[Tuple[int,int]]] = None,
-    return_exact_agents: bool = False
 ):
     agent_states = []
     agent_properties = []
-    recurrent_states = []
-
-    region_agent_remove_map = {i: [] for i in range(len(regions))}
-
-    if region_map is not None:
-        for (region_id, agent_id) in region_map:
-            try:
-                agent_states.append(regions[region_id].agent_states[agent_id])
-                agent_properties.append(regions[region_id].agent_properties[agent_id])
-                recurrent_states.append(regions[region_id].recurrent_states[agent_id])
-
-                region_agent_remove_map[region_id].append(agent_id)
-            except IndexError as e: 
-                exception_message = f"Warning: Unable to fetch specified agent ID {agent_id} in region {region_id}."
-                if not return_exact_agents: 
-                    iai.logger.debug(exception_message)
-                else:
-                    raise InvertedAIError(message=exception_message)
-    
+
     for ind, region in enumerate(regions):
         if not ind == exclude_index:
             if nearby_region is not None:
                 if sqrt((nearby_region.center.x-region.center.x)**2+(nearby_region.center.y-region.center.y)**2) > (REGION_MAX_SIZE + AGENT_SCOPE_FOV_BUFFER):
                     continue
             region_agent_states = region.agent_states
-            agent_states = agent_states + [state for i, state in enumerate(region_agent_states) if i not in region_agent_remove_map[ind]]
-            agent_properties = agent_properties + [prop for i, prop in enumerate(region.agent_properties[:len(region_agent_states)]) if i not in region_agent_remove_map[ind]]
-            recurrent_states = recurrent_states + [recurr for i, recurr in enumerate(region.recurrent_states) if i not in region_agent_remove_map[ind]]
-    
-    return agent_states, agent_properties, recurrent_states
+            agent_states = agent_states + [state for state in region_agent_states]
```

Comment:
why not just `agent_states + region_agent_states` ?

Reply:
Good catch, this was a leftover from splitting an earlier function into 2. I'll make this change.